### PR TITLE
Tests for links

### PR DESCRIPTION
**Pull request by joehoyle**
We currently don't have good coverage for links in all the responses, this is making it difficult to change areas
of code that affect links.

### CODE REVIEW

**Comment by joehoyle** [comment]
@WP-API/amigos #reviewmerge

File: tests/class-wp-test-rest-post-type-controller-testcase.php [hunk 1]
```diff
@@ -153,6 +153,48 @@ protected function check_post_data( $post, $data, $context ) {
 			sort( $data[ $taxonomy->rest_base ] );
 			$this->assertEquals( $terms, $data[ $taxonomy->rest_base ] );
 		}
+
+		// test links
+		if ( $links ) {
```
Comment (danielbachhuber) [inline tests/class-wp-test-rest-post-type-controller-testcase.php:158]: Are `$links` always passed?

Comment (danielbachhuber) [inline tests/class-wp-test-rest-post-type-controller-testcase.php:158]: ... meaning, is there some way we can always test them? I'm not a huge fan of conditionals inside of unit tests because I think it can be easy to mess up some of your logic and not notice that your tests are no longer running.

File: tests/test-rest-comments-controller.php [hunk 2]
```diff
@@ -1288,6 +1288,11 @@ protected function check_comment_data( $data, $context ) {
 		$this->assertEquals( mysql_to_rfc3339( $comment->comment_date_gmt ), $data['date_gmt'] );
 		$this->assertEquals( get_comment_link( $comment ), $data['link'] );
 		$this->assertContains( 'author_avatar_urls', $data );
+		$this->assertEqualSets( array(
+			'self',
+			'collection',
+			'up',
+		), array_keys( $links ) );
```
Comment (danielbachhuber) [inline tests/test-rest-comments-controller.php:1295]: Do we need to replicate this test in more controllers?

File: tests/class-wp-test-rest-post-type-controller-testcase.php (hunk 1 above)
Comment (joehoyle) [inline tests/class-wp-test-rest-post-type-controller-testcase.php:158]: @danielbachhuber this is due to the fact that items in collections don't have the same links in them as getting the single item, that's to say, getting a single item via `dispatch` will return a REST_Response that you call `get_links()` on, however that's a slightly different format to an collection item's `_links` property.

File: tests/test-rest-comments-controller.php (hunk 2 above)
Comment (joehoyle) [inline tests/test-rest-comments-controller.php:1295]: I'd rather get these merged, as this PR is only adding tests, these are better than none
